src/pool-bin/libraries/BinHooks.sol

```diff
@@ -163,9 +163,11 @@ library BinHooks {
         hookDeltaUnspecified += beforeSwapDelta.getUnspecifiedDelta();
 
         if (hookDeltaUnspecified != 0 || hookDeltaSpecified != 0) {
+            // the hookDelta is the delta added into amountIn which is in the opposite direction of the balanceDelta
+            // i.e. in a reasonable case, amountIn will basiaclly equal to "-BalanceDelta.amount0()/amount1()"
```

Comment:
This is now completely opposite when comparing with cl-pool because for cl-pool when `amountSpecified > 0` it means we are specifying amountOut, while for bin-pool we can only specify amountIn and it's always positive.

Reply:
want to clarify: does the hook for both Bin and CL have the same behavour for their return value? eg. `BeforeSwapDelta in beforeSwap` and `int128 at afterSwap()`?

Reply:
is this right? 

> 1/ what does negative BeforeSwapDelta for both value means?

1. if beforeSwapDelta is both negative, it means the hook want to give user extra tokenIn and tokenOut
2. if beforeSwapDelta is positive, it means the hook want to take a "fee" in both tokenIn and tokenOut 
3. likewise if its (positive , negative) it means hook want to take tokenIn as fee and give extra tokenOut

> 2/ what does negative int128 in afterSwap means?

1. if int128 is negative, it means the hook want to give amountOut 
2. if int128 is positive, it means the hook want to take some amountOut as "fee"

Reply:
for bin-pool, I think it should be opposite.
1. if beforeSwapDelta is both negative, it means the hook want to take a "fee" in both tokenIn and tokenOut
2. if beforeSwapDelta is positive, it means the hook want to give user extra tokenIn and tokenOut
3. likewise if its (positive , negative) it means hook want to give extra tokenIn and take tokenOut as fee

Let's elaborate case 3 for example:

if `BeforeSwapDelta.getSpecifiedDelta() > 0`, i.e. amountToSwap is even bigger than trader's specification, so more amountIn needed for this swap, but let's see who pay for the extra part:

from traders perspective:

The amountIn delta returned from `pool.swap` is definitely bigger in abs value because amountIn is bigger, but remember the delta is updated through `delta - hookDelta` in `afterSwap`

and we know `hookDelta = (-hookDeltaSpecified, -hookDeltaUnspecified)`
a negative value minus another negative value will result in a bigger negative value i.e. user are actually not paying that part


from hook perspective:

```
if (hookDelta != BalanceDeltaLibrary.ZERO_DELTA) {
    vault.accountPoolBalanceDelta(key, hookDelta, address(key.hooks));
}
```

This will count a negative amountIn delta on hook, so inside hook callback there must be a payment to the pool

Reply:
2/ what does negative int128 in afterSwap means?

Also, should be opposite 
if int128 is positive, it means the hook want to give amountOut
if int128 is negative, it means the hook want to take some amountOut as "fee"

In fact 
int128 in afterSwap  and `BeforeSwapDelta.getUnspecifiedDelta()` in beforeSwap are actually specifying the same thing.. 

can see they are combined in afterSwap before generating the final hookDelta: 

![CleanShot 2024-05-28 at 15 14 01@2x](https://github.com/pancakeswap/pancake-v4-core/assets/137024020/e7293f58-f86b-4ca8-b7de-2bef64867e1b)